internal/pkg/cli/task_run.go

```diff
@@ -259,6 +264,9 @@ func (o *runTaskOpts) Ask() error {
 	if err := o.askTaskGroupName(); err != nil {
 		return err
 	}
+	if err := o.askAppName(); err != nil {
+		return err
+	}
```

Comment:
I think this looks good, the idea behind `askAppName` is to allow users to run their task in a Copilot environment even outside of their workspace. 
To be able to ask which env they want to run in we need to prompt for the application name.

Reply:
If a user doesn't specify the appname, but they're in an existing environment, do we want to list every application that they have in their account? 

Doesn't it make more sense to use the app of that workspace? I'm wonder if this would lead to more confusion? Having a task that's running in one app's workspace, being run in another app's environment (if they're running a bunch of these in a single app's workspace to identify which one is running where)?

Reply:
Exactly! We prompt for app only if the `--app` flag isn't specified and that they are not under any workspace. This logic is written in the `askAppname()` method, not in `Ask()`.